frontends/web/src/routes/new-settings/components/advanced-settings/tor-proxy-dialog.tsx

```diff
@@ -0,0 +1,102 @@
+import { Dispatch, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Dialog, DialogButtons } from '../../../../components/dialog/dialog';
+import { Toggle } from '../../../../components/toggle/toggle';
+import { Button, Input } from '../../../../components/forms';
+import { setConfig } from '../../../../utils/config';
+import { socksProxyCheck } from '../../../../api/backend';
+import { alertUser } from '../../../../components/alert/Alert';
+
+type TProps = {
+  open: boolean;
+  config: any;
+  onCloseDialog: () => void;
+  onChangeConfig: (config: any) => void;
+  handleShowRestartMessage: Dispatch<boolean>;
+}
+
+export const TorProxyDialog = ({ open, config, onCloseDialog, onChangeConfig, handleShowRestartMessage }: TProps) => {
+  const [proxyAddress, setProxyAddress] = useState<string>();
+  const [proxyEnabled, setProxyEnabled] = useState<boolean>();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    if (config && config.backend) {
+      setProxyAddress(config.backend.proxy.proxyAddress);
+      setProxyEnabled(config.backend.proxy.useProxy);
+    }
+  }, [config]);
+
+
+  const handleSetProxyButton = async () => {
+    if (!config || proxyAddress === undefined) {
+      return;
+    }
+    const proxy = config.backend.proxy;
+    proxy.proxyAddress = proxyAddress.trim();
+
+    const result = await socksProxyCheck(proxy.proxyAddress);
+    const { success, errorMessage } = result;
+
+    if (success) {
+      setProxyConfig(proxy);
+    } else {
+      alertUser(errorMessage || t('account.fatalError'));
+    }
+  };
+
+  const setProxyConfig = async (proxyConfig: any) => {
```

Comment:
config layer is not typed, but I wonder if it would make sense to add minimal type for proxyConfig in this component.

Reply:
> also I am wondering if it is really needed to receive updates via onChangeConfig and pass config down again... but no blocker. keep as is or feel free to change.

Keeping would be good IMO, reason: https://github.com/digitalbitbox/bitbox-wallet-app/pull/2104#discussion_r1190496329 ,


 another additional reason why we need to update this config state (that lives within the tor dialog), is bc we're also using this `config` state in the `EnableTorProxySetting` (which displays "Disabled" and "Enabled - therefore requires an updated version of config).

Reply:
> not important rightnow but what do you think of, instead of passing the whole config to the components, only pass the relevant part?


Yeah we can definitely find out which parts of config we're supposed to be passing from the component that's using it and just pass exactly those. Would be a bit tricky / more guesswork right now as config is not typed for now.. but yh we can do this:)